src/mscorlib/shared/System/Text/StringBuilder.cs

```diff
@@ -2001,6 +2043,22 @@ private static unsafe void ThreadSafeCopy(char* sourcePtr, char[] destination, i
             }
         }
 
+        private static unsafe void ThreadSafeCopy(char* sourcePtr, Span<char> destination, int destinationIndex, int count)
```

Comment:
My take was that there is necessarily some overhead, since we would add the conversion from array to span on top of the existing code. But it's true that this overhead is probably not significant in this context. I'll run some benchmark to make sure.